sw/device/silicon_creator/manuf/ft.c

```diff
@@ -0,0 +1,52 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "sw/device/lib/base/status.h"
+#include "sw/device/lib/dif/dif_flash_ctrl.h"
+#include "sw/device/lib/dif/dif_lc_ctrl.h"
+#include "sw/device/lib/dif/dif_otp_ctrl.h"
+#include "sw/device/lib/testing/test_framework/check.h"
+#include "sw/device/lib/testing/test_framework/ottf_main.h"
+#include "sw/device/silicon_creator/manuf/lib/provisioning.h"
+
+#include "flash_ctrl_regs.h"  // Generated
+#include "hw/top_earlgrey/sw/autogen/top_earlgrey.h"
+#include "lc_ctrl_regs.h"   // Generated
+#include "otp_ctrl_regs.h"  // Generated
+
+OTTF_DEFINE_TEST_CONFIG();
+
+/**
+ * DIF Handles.
+ *
+ * Keep this list sorted in alphabetical order.
+ */
+static dif_flash_ctrl_state_t flash_state;
+static dif_lc_ctrl_t lc_ctrl;
+static dif_otp_ctrl_t otp_ctrl;
+
+/**
+ * Initializes all DIF handles used in this module.
+ */
+static status_t peripheral_handles_init(void) {
+  TRY(dif_flash_ctrl_init_state(
+      &flash_state,
+      mmio_region_from_addr(TOP_EARLGREY_FLASH_CTRL_CORE_BASE_ADDR)));
+  TRY(dif_lc_ctrl_init(mmio_region_from_addr(TOP_EARLGREY_LC_CTRL_BASE_ADDR),
+                       &lc_ctrl));
+  TRY(dif_otp_ctrl_init(mmio_region_from_addr(TOP_EARLGREY_OTBN_BASE_ADDR),
+                        &otp_ctrl));
+  return OK_STATUS();
+}
+
+bool test_main(void) {
+  CHECK_STATUS_OK(peripheral_handles_init());
+
+  // TODO: This test is a temporary placeholder which will get replaced for an
+  // interactive test loop once we have host side instrumentation available.
+  CHECK_STATUS_OK(
+      provisioning_device_secrets_start(&flash_state, &lc_ctrl, &otp_ctrl));
+
```

Comment:
going to call `provisioning_device_secrets_end()` here? or still adding stuff?

Reply:
yeah, still adding stuff. I will add a device reset and then call the end function.